public/app/features/dashboard-scene/saving/SaveProvisionedDashboard.tsx

```diff
@@ -0,0 +1,134 @@
+import { css } from '@emotion/css';
+import { useEffect, useMemo, useState } from 'react';
+
+import { config, getBackendSrv } from '@grafana/runtime';
+import { Button, Stack, Box, TextArea, Field, Input, Alert } from '@grafana/ui';
+import { AnnoKeyRepoName, AnnoKeyRepoPath } from 'app/features/apiserver/types';
+import { DashboardMeta } from 'app/types';
+
+import { DashboardScene } from '../scene/DashboardScene';
+
+import { SaveDashboardDrawer } from './SaveDashboardDrawer';
+import { SaveDashboardFormCommonOptions } from './SaveDashboardForm';
+import { DashboardChangeInfo } from './shared';
+
+export interface Props {
+  meta: DashboardMeta;
+  dashboard: DashboardScene;
+  drawer: SaveDashboardDrawer;
+  changeInfo: DashboardChangeInfo;
+}
+
+export function SaveProvisionedDashboard({ meta, drawer, changeInfo }: Props) {
+  const dashboardJSON = useMemo(() => JSON.stringify(changeInfo.changedSaveModel, null, 2), [changeInfo]);
+
+  const [repo, setRepo] = useState<string>();
+  const [path, setPath] = useState<string>();
+  const [ref, setRef] = useState<string>();
+  const [comment, setComment] = useState<string>();
+
+  useEffect(() => {
+    const anno = meta.k8s?.annotations;
+    if (!anno) {
+      setRepo('');
+      setPath('');
+      setRef('');
+      return;
+    }
+    let ref = '';
+    let path = anno[AnnoKeyRepoPath] ?? '';
+    const idx = path.indexOf('#');
+    if (idx > 0) {
+      ref = path.substring(idx + 1);
+      path = path.substring(0, idx);
+    }
+    setRepo(anno[AnnoKeyRepoName]);
+    setPath(path);
+    setRef(ref);
+  }, [meta]);
+
+  const doSave = () => {
+    const url = `apis/provisioning.grafana.app/v0alpha1/namespaces/${config.namespace}/repositories/${repo}/files/${path}`;
```

Comment:
We'd be able to use hooks here from the endpoints file, but I'll add those in a separate PR.